### PR TITLE
Update install docs for architecture build options.

### PR DESCRIPTION
**Pull request by halfflat**

### CODE REVIEW

File: doc/install.rst [hunk 1]
```diff
@@ -251,46 +257,27 @@ with ``-g -O0`` flags), use the ``CMAKE_BUILD_TYPE`` CMake parameter.
...
-Vectorization
--------------
+Vectorization and architecture
+------------------------------
 
-Explicit vectorization of key computational kernels can be enabled in Arbor by setting the
-``ARB_VECTORIZE_TARGET`` CMake parameter:
+For both optimization and vectorization, the target architecture for the library
+can be specified with ``ARB_ARCH``. Values correspond to the architecture names
+given to (for example) GCC and Clang with the ``-mcpu`` or ``-march`` options.
```
[the remark this answers is not in the document]
Comment (halfflat) [reply doc/install.rst]: I'll split architecture section from vectorization. I can mention a subset of the possible `-mcpu`/`-march` options, but it really just corresponds to whatever gcc supports (plus some renaming for a subset for other compilers).

File: doc/install.rst [hunk 2]
```diff
@@ -251,46 +257,27 @@ with ``-g -O0`` flags), use the ``CMAKE_BUILD_TYPE`` CMake parameter.
...
+.. code-block:: bash
 
-    .. code-block:: bash
+    cmake -DARB_VECTORIZE=ON
 
-        cmake -DCMAKE_BUILD_TYPE=release \
-              -DARB_ARCH=broadwell       \
-              -DARB_VECTORIZE=ON         \
+By default, these kernels are not explicitly vectorized and will instead rely
+on compiler auto-vectorization.
```
[the remark this answers is not in the document]
Comment (halfflat) [reply doc/install.rst]: Do we need to state which vectorization ISA is used for which architecture? We're not exposing this information anywhere in the standard build interface. We could just list those architectures for which explicit vectorization will use ISA-specific intrinsics.

**Comment by halfflat** [comment]
Update PR to address review points.

File: doc/install.rst [hunk 3]
```diff
@@ -257,28 +257,44 @@ with ``-g -O0`` flags), use the ``CMAKE_BUILD_TYPE`` CMake parameter.
...
-Explicit vectorization of key computational kernels can be enabled in Arbor by setting the
-``ARB_VECTORIZE_TARGET`` CMake flag:
+When deploying on a different machine, one should, for an optimized library, specify
+the specific architecture of that machine. The valid values correspond to those given
+to the ``-mcpu`` or ``-march`` options for GCC and Clang; the build system will translate
+these names to corresponding values for other supported compilers.
+
+Specific recent x86-family CPU architectures include ``ivybridge``, ``haswell``,
+``broadwell``, ``skylake``, ``knl`` (Intel) and ``bdver1``, ``bdver2``, ``bdver3``,
+``bdver4``, ``znver1`` (AMD).
```
Comment (bcumming) [inline doc/install.rst]: We haven't tested on AMD, so I wouldn't mention it!

File: doc/install.rst [hunk 4]
```diff
@@ -257,28 +257,44 @@ with ``-g -O0`` flags), use the ``CMAKE_BUILD_TYPE`` CMake parameter.
...
 .. code-block:: bash
 
     cmake -DARB_VECTORIZE=ON
 
-By default, these kernels are not explicitly vectorized and will instead rely
-on compiler auto-vectorization.
-
+With this flag set, the library will use architecture-specific vectorization intrinsics
+to implement these kernels. Supported vectorization ISAs are currently limited to
+AVX, AVX2 and AVX512 as supported on recent Intel and AMD x86 family CPUs.
```
Comment (bcumming) [inline doc/install.rst]: How to mention AVX, AVX2 and AVX512? The user can specify an architecture, e.g. `native` or `haswell`, not an ISA.
Maybe mention which architectures will work when `ARB_VECTORIZE` is turned on.
Or state that the `ARB_ARCH` must support one of the ISAs, with an example/list of arch that we have tested it on.

**Comment by halfflat** [comment]
These have to be radically updated for the new split build / installable target stuff.

File: doc/install.rst [hunk 5]
```diff
@@ -500,24 +503,37 @@ Troubleshooting
...
+    On systems where the build node and compute node have different architectures
+    within the same family, this may mean that separate compilation of ``modcc``
+    is not necessary.
+
 .. Warning::
     ``Illegal instruction`` errors are a sure sign that
     Arbor is running on a system that does not support the architecture it was compiled for.
 
-When cross compiling, we have to take care that the *modcc* compiler, which is used to convert NMODL to C++/CUDA code, is able to run on the compilation node.
+When cross compiling, we have to take care that the *modcc* compiler, which is
```
[the remark this answers is not in the document]
Comment (halfflat) [reply doc/install.rst]: Yep, this is now set with `ARB_MODCC` at configure time. Set ⇒ use that binary; Unset ⇒ build and use in-tree modcc.

`modcc` and `unit-modcc` are valid targets either way, but they will not be built by default if `ARB_MODCC` is set.

This has to all go into the updated documentation (which I will do!).

**Comment by halfflat** [comment]
Updated PR to address issue #532, as well as comments regarding architecture names.

**Pull request merged**